### DISCUSSION TITLE
How to install on Linux desktop/tablet/phone?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [OpenVoiceOS/ovos-buildroot#181] [Q&A] [answered]
I'm utterly new to the project, and obviously joined at a bad time. Consequently, the documentation appears a little disparate, sparse, and unhelpfully concise. However, I doubt that this should significantly negatively affect the quality of the software based upon your press release at https://community.mycroft.ai/t/faq-ovos-neon-and-the-future-of-the-mycroft-voice-assistant/13496?u=rokejulianlockhart, so I'm trying to learn how to use it anyway.

I see that https://github.com/OpenVoiceOS/ovos-core#ovos-core states that to install, one performs merely

> `pip install ovos-core[mycroft]`

However, for what *appears* to be server-like software similar to Nextcloud (server, not client) this ease of installation surprises me. Considering that https://openvoiceos.github.io/community-docs/gs_installation_choices/ doesn't mention this method whatsoever, I am doubtful. Additionally, whether https://github.com/OpenVoiceOS/ovos-gui#ovos-gui-messagebus is included with this command isn't specified, nor is whether the user can easily uninstall the package (should it not work or contain what they expect).

My hope is that I've misinterpreted this software's architecture, especially since OVOS is meant to be (and is, from cursory inspection) more modular than Mycroft's implementation was. However, is it really installable and executable like https://play.google.com/store/apps/details?id=com.google.android.apps.googleassistant&hl=en&gl=US and https://apps.microsoft.com/store/detail/cortana/9NFFX4SZZ23L are? Like someone just wanting a digital assistant for their Linux PC, tablet or phone would want?

**Comment by JarbasAl** [comment, marked as answer]
Hello and welcome to the community!

To explain the architecture let's start from how mycroft-core worked and see how things evolved since 

- mycroft-core was a single repository including the full voice assistant software
- even if it was a single repo, mycroft-core was architected as a few different independent services
- messagebus, skills, audio playback, speech, enclosure, debug_cli
- these were hard to test and improve on because they all imported code from everywhere else in mycroft-core
- since it was so big it was hard to package properly for multiple platforms
- bash script to launch and update everything

with ovos we did the following / is happening for release 0.0.8
- services have been extracted into individual repos with their own release cycles and unittests
- services communicate exclusively via messagebus
- each service is optional, eg, a webchat won't need audio 
- each service is packages and published on pypi
- each service provides a entry point when installed, just launch it from the cli
- each service can be easily replaced by alternate implementations
   -  default listener now is ovos-dinkum-listener, an improved version taken from the mk2
   -  we have proof of concept c++ and rust implementations for some services
   
the documentation is very bare bones and doesn't account for a lot of changes coming in 0.0.8 release  (dev branch), apologies!

the `pip install ovos-core[mycroft]` is dragging all the requirements for a classic mycroft install, each service is in it's requirements.txt , this is only meant to help mycroft users moving over, if possible one should install/package/launch each service individually

**Reply by roke-julian-lockhart** [reply]
Thanks lots. If you didn't copy that from somewhere, I promise I appreciate the work put into that list. Even if you did, I couldn't find it!

For a layman like myself (or a novice software developer at best) is this software usable as a local assistant, for a smartphone or desktop system? I have a Linux phone running PMOS, and a desktop and tablet both running OpenSUSE TW. I also have a laptop with Win 11. I'd love to be able to use a single, cross-platform VA rather than just Copilot on W11 and nothing on the others.

Does OVOS provide this? I've no issue running a server backend like NC needs, as long as I can just use it like any other application.

**Reply by builderjer** [reply]
We really apologize for lack of documentation,  they ARE being updated, it's just been a while so there is a lot of changes. 

What you are looking for is in heavy development right now. We, @JarbasAl calls it,  [hive-mind](https://github.com/JarbasHiveMind/HiveMind-core).  The docs are really lacking there, but once it is ready, I'm sure they will improve. 

There is also a [test image](https://ovosimages.ziggyai.online/raspbian/hivemind/voicesat/) for a 64b rpi

This software does work,  but may change daily

**Reply by roke-julian-lockhart** [reply]
Thanks lots, @builderjer. I see in the ReadMe that it mentions that OVOS is usually installed on desktops. Whilst waiting for https://github.com/JarbasHiveMind/HiveMind-core to become somewhat more stable, is the aforementioned

```python3
pip install ovos-core[mycroft]
```

all that I need to do in order to install it? If so, that's wonderfully convenient.

**Reply by builderjer** [reply]
That will install all of the libraries, but will not start the modules.  There is a **WIP** startup script that may work on a x86 ubuntu machine.  
https://github.com/OpenVoiceOS/raspbian-ovos/blob/dev/manual_user_install.sh
Let me know if it works for you
You can also look at the code and check the systemd files for examples of how to start the modules.

There is also a [PR](https://github.com/OpenVoiceOS/community-docs/pull/68) for the community docs with a whole rewrite.

The relevant section would be [here](https://github.com/OpenVoiceOS/community-docs/blob/update/docs/install_ovos_core.md) or [here](https://github.com/OpenVoiceOS/community-docs/blob/update/docs/starting_systemd.md)

Any input on that PR or the [update branch](https://github.com/OpenVoiceOS/community-docs/tree/update) in particular is appreciated also

**Reply by roke-julian-lockhart** [reply]
@builderjer, do you think I could make an issue about a comprehensive automated installation method so that I can be notified when one is available (via the issue closure)? I imagine that others would be interested, considering that it's a fairly significant part of any software, and utterly necessary for any kind of widespread adoption.

**Reply by JarbasAl** [reply]
the goal really is that `pip install ovos-core` is the only step needed by default, anything after that would be considered advanced usage and optional

what would you expect from such script? it seems to be most users get confused with how to launch ovos after the install step, but i'm not sure we want to force auto start either and this seems more a documentation issue and also platform specific

we no longer need a setup skill since dropping support for selene, so there is some discussion here https://github.com/OpenVoiceOS/skill-ovos-setup/issues/76 to make an installer for friendly user facing configuration

**Reply by roke-julian-lockhart** [reply]
@JarbasAl,

> what would you expect from such script? it seems to be most users get confused with how to launch ovos after the install step, but i'm not sure we want to force auto start either and this seems more a documentation issue and also platform specific

If a dialog window and/or message in the terminal (if installed via that manner) appeared with a hyperlink to the relevant documentation, that'd be a good start. I don't mind following instructions, even if they're complex, I just want to always know exactly which to follow and when.

> what would you expect from such script? it seems to be most users get confused with how to launch ovos after the install step, but i'm not sure we want to force auto start either

I'd have it auto-start upon install and boot if I were you, since I'd consider any use-case in which a user wants this installed but doesn't want that to be fairly weird. But if it's just a few `systemctl` commands to fix that, I don't care much either way.

> there is some discussion here https://github.com/OpenVoiceOS/skill-ovos-setup/issues/76 to make an installer for friendly user facing configuration

That's exactly what I was looking for. That issue really should be renamed.

---

~~I'll go test `pip install ovos-core` now. Wish me luck!~~

https://github.com/OpenVoiceOS/community-docs/blob/update/docs/install_ovos_core.md is definitely better than https://github.com/OpenVoiceOS/ovos-core#ovos-core. I'll follow its instructions.

**Reply by builderjer** [reply]
> https://github.com/OpenVoiceOS/community-docs/blob/update/docs/install_ovos_core.md is definitely better than https://github.com/OpenVoiceOS/ovos-core#ovos-core.

That PR is being proofread now and should be merged soon.  After it is, please open issues on where it needs to be improved.
